### ISSUE TITLE
Gridcoin rpc for a row transaction not returning the hex value of a ScriptPuKey

### ISSUE THREAD

**Issue by grctest** [gridcoin-community/Gridcoin-Research#86]
Source of issue: https://www.reddit.com/r/gridcoin/comments/4gu6ao/gridcoin_rpc_for_a_row_transaction_not_returning/

> Hello community,
> I am on the coinvault.io web wallet team.
> I noticed the Gridcoin client rpc method GetRawTransaction is not retuning the HEX value of the script public key only the ASM.
> We've worked around it for now but Is there a reason for that, will the devs be able to change that in the next release?

Location of issue: https://github.com/gridcoin/Gridcoin-Research/blob/master/src/rpcrawtransaction.cpp#L282

**Comment by gridcoin** [comment]
Yes, fixed.  It will be in the next release.

Rob

**Comment by gridcoin** [comment]
This is what you wanted, right ?:

13:35:30
￼
getrawtransaction 2ab6c77027cd887e48c29a69017c13cf113a558e56b5c5bccc37fa126cea3e49 1

13:35:30
￼
{
"hex" : "01000000d09323570187471608a334dbd2003caef1684ffe0208483df441db8c9a0dcd900783560fad0100000048473044022042692a584f333a3778157bb8b27d566c1a2febf27f5780c569fc57ae057cd3f0022021b0aa9cca8cc1931e46581feeb28fbaac2f16c02a37989b9a3e11029c8ab4a901ffffffff0300000000000000000070f0d6bbd8e400002321027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6ac40420f00000000002321027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6ac0000000000",
"txid" : "2ab6c77027cd887e48c29a69017c13cf113a558e56b5c5bccc37fa126cea3e49",
"version" : 1,
"time" : 1461949392,
"locktime" : 0,
"hashboinc" : "",
"vin" : [
{
"txid" : "ad0f56830790cd0d9a8cdb41f43d480802fe4f68f1ae3c00d2db34a308164787",
"vout" : 1,
"scriptSig" : {
"asm" : "3044022042692a584f333a3778157bb8b27d566c1a2febf27f5780c569fc57ae057cd3f0022021b0aa9cca8cc1931e46581feeb28fbaac2f16c02a37989b9a3e11029c8ab4a901",
"hex" : "473044022042692a584f333a3778157bb8b27d566c1a2febf27f5780c569fc57ae057cd3f0022021b0aa9cca8cc1931e46581feeb28fbaac2f16c02a37989b9a3e11029c8ab4a901"
},
"sequence" : 4294967295
}
],
"vout" : [
{
"value" : 0.00000000,
"n" : 0,
"scriptPubKey" : {
"asm" : "",
"hex" : "",
"type" : "nonstandard"
}
},
{
"value" : 2516195.15494512,
"n" : 1,
"scriptPubKey" : {
"asm" : "027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6 OP_CHECKSIG",
"hex" : "21027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6ac",
"reqSigs" : 1,
"type" : "pubkey",
"addresses" : [
"S2bWE7QQP3xujqA8ejnNKr6zuMVNECqHVP"
]
}
},
{
"value" : 0.01000000,
"n" : 2,
"scriptPubKey" : {
"asm" : "027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6 OP_CHECKSIG",
"hex" : "21027de0afade5e2df9cea23c4f10e6539f4433724cf517ca50da333437ce611b3a6ac",
"reqSigs" : 1,
"type" : "pubkey",
"addresses" : [
"S2bWE7QQP3xujqA8ejnNKr6zuMVNECqHVP"
]
}
}
],
"blockhash" : "1ef4fc1e3d95dce2cde5b78a6a1f6c972f9101b4aa5cdb5605412548677350be",
"confirmations" : 959,
"time" : 1461949392,
"blocktime" : 1461949392
}

**Comment by dangershony** [comment]
Thanks that's exactly what I needed.
Out of curiosity was there any particular reason the HEX was not populated?

**Comment by gridcoin** [comment]
Good - glad it will work, it should be out any day now.   Ill be working on some of these tickets tonight.

Uhhhh- it was not intentional - as a matter of fact - I dont know why it was left out.  But to make an educated guess, we used to have a prior version of Gridcoin for Scrypt (before we moved to Gridcoin-research).  Some of the subs were ported to Research.  We might have used that particular old RPC call instead of the stake call.  

Rob
